compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3466,6 +3466,28 @@ boolean checkLiteralAssignabilityBasedOnType(BLangLiteral baseLiteral, BLangLite
         return false;
     }
 
+    boolean isFloatLiteralValue(Location pos, String numericLiteral) {
+        double value = Double.parseDouble(String.valueOf(numericLiteral));
+        if (Double.isInfinite(value)) {
+            dlog.error(pos, DiagnosticErrorCode.FLOAT_TOO_LARGE, numericLiteral);
+            return false;
+        }
+
+        if (value == 0.0) {
+            for (int i = 0; i < numericLiteral.length(); i++) {
```

Comment:
Currently, we represent float values not included in the float range as `infinity`(value > Max_Value) and `0.0`(value < Min_Value). With this change, we give an error for those values.

Reply:
I was wondering when the parsed value will be "0.0" but not small. The user-provided "0.0" literal case. Shall we try to handle this using a regex maybe?

Reply:
Also, it would be clearer to do this check at the very beginning anyway? IMO, at compile time `0.0` can be quite a common case for floating point literals so we won't be losing much by doing this check directly on `numericLiteral`.